eclipse-jdt/eclipse.jdt.core
# List of patterns in case statements

- Parse multiple Patterns in a case statement
- AST & generator changes for list of patterns
- Fix error reporting for a pattern variable + multiple patterns

Closes #1643



## What it does

Parse case statements with multiple patterns, as per JEP 443.

## How to test

Confirm the following code snippet compiles and executes as expected:

```java
Object o = "Hello, World";
switch (o) {
case String _, List _ when 1 == 1: System.out.println("it's working"); break;
default: System.out.println("it's not working");
}
```

Confirm that errors are placed on `a` and `b` in the following snippet:

```java
Object o = "Hello, World";
switch (o) {
case String a, List b when 1 == 1: System.out.println("it's working"); break;
default: System.out.println("it's not working");
}
```

## Author checklist
